### PR TITLE
KAFKA-18189: CoordinatorRequestManager log message can include incorrect coordinator disconnect time

### PR DESCRIPTION
**Pull request by kirktrue**
Fixed logic in `markCoordinatorUnknown()` to ensure the warning log contains the correct number of milliseconds the client has been disconnected.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: clients/src/test/java/org/apache/kafka/clients/consumer/internals/CoordinatorRequestManagerTest.java [hunk 1]
```diff
@@ -75,6 +83,65 @@ public void testSuccessfulResponse() {
...
+            coordinatorRequestManager.markCoordinatorUnknown("test", time.milliseconds());
+            assertMillisecondEquals(appender, oneMinute * 2);
+        }
+    }
+
+    private void assertMillisecondEquals(LogCaptureAppender appender, long expected) {
+        List<Long> ms = millisFromLog(appender);
+        assertFalse(ms.isEmpty());
+        long actual = ms.get(ms.size() - 1);
+        assertEquals(expected, actual, "The ");
```
Comment (clolov) [inline clients/src/test/java/org/apache/kafka/clients/consumer/internals/CoordinatorRequestManagerTest.java]: For my understanding, is the message upon assertion failure meant to be just "The"?

File: clients/src/test/java/org/apache/kafka/clients/consumer/internals/CoordinatorRequestManagerTest.java [hunk 2]
```diff
@@ -75,6 +83,65 @@ public void testSuccessfulResponse() {
...
+     * disconnect logic, controlling the logging and system time, to ensure the warning message is correct.
+     *
+     * @see CoordinatorRequestManager#markCoordinatorUnknown(String, long)
+     */
+    @Test
+    public void testMarkCoordinatorUnknownLoggingAccuracy() {
+        long oneMinute = 60000;
+
+        try (final LogCaptureAppender appender = LogCaptureAppender.createAndRegister()) {
+            appender.setClassLogger(CoordinatorRequestManager.class, Level.DEBUG);
```
Comment (clolov) [inline clients/src/test/java/org/apache/kafka/clients/consumer/internals/CoordinatorRequestManagerTest.java:111]: Also for my understanding, if the message is at INFO level why do you set the level to DEBUG here?

File: clients/src/test/java/org/apache/kafka/clients/consumer/internals/CoordinatorRequestManagerTest.java (hunk 1 above)
Comment (kirktrue) [reply clients/src/test/java/org/apache/kafka/clients/consumer/internals/CoordinatorRequestManagerTest.java]: Will fix.

File: clients/src/test/java/org/apache/kafka/clients/consumer/internals/CoordinatorRequestManagerTest.java (hunk 2 above)
Comment (kirktrue) [reply clients/src/test/java/org/apache/kafka/clients/consumer/internals/CoordinatorRequestManagerTest.java:111]: Will fix.

Comment (kirktrue) [reply clients/src/test/java/org/apache/kafka/clients/consumer/internals/CoordinatorRequestManagerTest.java:111]: Double-checking now, the warning message is logged at `DEBUG` in [`CoordinatorRequestManager.java`](https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/clients/consumer/internals/CoordinatorRequestManager.java#L158), so I believe the code is correct as-is. PLMK if I'm misunderstanding. Thanks!

File: clients/src/test/java/org/apache/kafka/clients/consumer/internals/CoordinatorRequestManagerTest.java (hunk 1 above)
Comment (kirktrue) [reply clients/src/test/java/org/apache/kafka/clients/consumer/internals/CoordinatorRequestManagerTest.java]: Removed errant error message

File: clients/src/test/java/org/apache/kafka/clients/consumer/internals/CoordinatorRequestManagerTest.java (hunk 2 above)
Comment (clolov) [reply clients/src/test/java/org/apache/kafka/clients/consumer/internals/CoordinatorRequestManagerTest.java:111]: Ah, I see, I was looking at the wrong log line to be captured. Thank you for double-checking!

**Review by clolov — APPROVED** [review]
Looks great, thank you!

**Pull request merged**